terminal.h

```diff
@@ -504,7 +503,7 @@ void codepoint_to_utf8(std::string &s, char32_t c) {
     } else {
         throw std::runtime_error("Invalid UTF32 codepoint.");
     }
-    char u1, u2, u3, u4;
+    char32_t u1('x'), u2('x'), u3('x'), u4('x');
```

Comment:
I think these need to be `char8_t`, which should be the same as `char`.

Reply:
It looks like 'char` is the correct type. This allows me to remove all of the static_casts. BTW - it looks like [char8_t](https://en.cppreference.com/w/cpp/keyword/char8_t) is a C++20 feature and so not available to my project yet.